activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -48,7 +42,9 @@ def sql(event)
       binds = nil
 
       unless (payload[:binds] || []).empty?
-        binds = "  " + payload[:binds].map { |attr| render_bind(attr) }.inspect
+        binds = "  " + payload[:binds].zip(payload[:type_casted_binds] || []).map { |attr, value|
```

Comment:
A collection should never be `nil`, a collection can at most be empty.

We assume `payload[:binds]` responds to `zip` (to `map` in the original code) and do not have a defensive `||` for it.